tests/kernels/test_kernels.py

```diff
@@ -0,0 +1,500 @@
+#         params = np.array([0.5, 0.9])
+
+#         val = k(0.1, 0.1, params)
+
+#         assert val == pytest.approx(1.0)
+
+#     def test_square_kernel_matrix(self):
+#         dev = qml.device("default.qubit", wires=1)
+#         k = kern.EmbeddingKernel(_simple_ansatz, dev)
+#         params = np.array([0.5, 0.9])
+
+#         K = k.square_kernel_matrix([0.1, 0.2, 0.4], params)
+
+#         # TODO: Add value tests
+
+#         assert K.shape == (3, 3)
+#         assert np.allclose(K, np.transpose(K))
+#         assert np.allclose(np.diag(K), np.array([1, 1, 1]))
+
+#     def test_kernel_matrix(self):
+#         dev = qml.device("default.qubit", wires=1)
+#         k = kern.EmbeddingKernel(_simple_ansatz, dev)
+#         params = np.array([0.5, 0.9])
+
+#         X1 = [0.1, 0.2, 0.4]
+#         X2 = [1.0, 2.1]
+
+#         K = k.kernel_matrix(X1, X2, params)
+
+#         # TODO: Add value tests
+
+#         assert K.shape == (3, 2)
+
+#         for i in range(3):
+#             for j in range(2):
+#                 assert K[i, j] == k(X1[i], X2[j], params)
+
+#     def test_kernel_target_alignment(self):
+#         dev = qml.device("default.qubit", wires=1)
+#         k = kern.EmbeddingKernel(_simple_ansatz, dev)
+#         params = np.array([0.5, 0.9])
+
+#         alignment = k.target_alignment([0.1, 0.2, 0.4], [1, -1, 1], params)
+
+#         # TODO: Add value tests
+
+#         assert 0 <= alignment
+#         assert alignment <= 1
+
+#     def test_kernel_polarization(self):
+#         dev = qml.device("default.qubit", wires=1)
+#         k = kern.EmbeddingKernel(_simple_ansatz, dev)
+#         params = np.array([0.5, 0.9])
+
+#         polarization = k.polarization([0.1, 0.2, 0.4], [1, -1, 1], params)
+
+#         # TODO: Add value tests
+
+#         assert 0 <= polarization
+
+
+def _mock_kernel(x1, x2, history):
+    history.append((x1, x2))
+
+    if x1 == x2:
+        return 1
+    else:
+        return 0.2
+
+
+def _laplace_kernel(x1, x2):
+    return np.exp(-math.fabs(x1 - x2))
+
+
+class TestKernelMatrix:
+    def test_simple_kernel(self):
+        X = [0.1, 0.4]
+
+        K_expected = np.array([[1, 0.2], [0.2, 1]])
+
+        K = kern.square_kernel_matrix(X, lambda x1, x2: _mock_kernel(x1, x2, []))
+
+        assert np.array_equal(K, K_expected)
+
+    def test_laplace_kernel(self):
+        X = [0.1, 0.4, 0.2]
+
+        K_expected = np.exp(-np.array([[0.0, 0.3, 0.1], [0.3, 0.0, 0.2], [0.1, 0.2, 0.0]]))
+
+        K = kern.square_kernel_matrix(X, _laplace_kernel, assume_normalized_kernel=False)
+
+        assert np.array_equal(K, K_expected)
+
+
+class TestKernelPolarization:
+    def test_correct_calls(self):
+        X = [0.1, 0.4]
+        Y = [1, -1]
+
+        hist = []
+
+        kern.kernel_polarization(X, Y, lambda x1, x2: _mock_kernel(x1, x2, hist))
+
+        assert len(hist) == 3
+
+        assert (0.1, 0.4) in hist
+        assert (0.1, 0.1) in hist
+        assert (0.4, 0.4) in hist
```

Comment:
Wouldn't it be more powerful to just check that `hist` is correct (which would also check the order)?